crates/chainio/txmanager/src/simple_tx_manager.rs

```diff
@@ -0,0 +1,265 @@
+        // TODO: It also takes care of gas estimation and adds a buffer to the gas limit
+        // TODO: Estimating gas and nonce
+        //m.log.Debug("Estimating gas and nonce")
+        //tx, err := m.estimateGasAndNonce(ctx, tx)
+        let signer = self.create_local_signer()?;
+        let _signed_tx = signer
+            .sign_transaction(tx)
+            .await
+            .map_err(|_| TxManagerError::SignerError)?;
+
+        self.logger.debug("Transaction signed", &[&tx]);
+
+        // send transaction and get receipt
+        let pending_tx = self
+            .provider
+            .send_transaction(tx.clone().into())
+            .await
+            .map_err(|_| TxManagerError::SendTxError)?;
+
+        self.logger
+            .debug("Transaction sent. Pending transaction: ", &[&pending_tx]);
+
+        // wait for the transaction to be mined
+        SimpleTxManager::wait_for_receipt(pending_tx).await
+    }
+
+    /// Waits for the transaction receipt.
+    ///
+    /// This is a wrapper around `PendingTransactionBuilder::get_receipt`.
+    ///
+    /// # Arguments
+    ///
+    /// - `pending_tx`: The pending transaction builder we want to wait for.
+    ///
+    /// # Returns
+    ///
+    /// - The block number in which the transaction was included.
+    /// - `None` if the transaction was not included in a block or an error ocurred.
+    pub async fn wait_for_receipt(
+        pending_tx: PendingTransactionBuilder<'_, Transport, Ethereum>,
+    ) -> Result<TransactionReceipt, TxManagerError> {
+        pending_tx
+            .get_receipt()
+            .await
+            .map_err(|_| TxManagerError::WaitForReceiptError)
+    }
+
+    /*
+    // GetNoSendTxOpts This generates a noSend TransactOpts so that we can use
+    // this to generate the transaction without actually sending it
+    func (m *SimpleTxManager) GetNoSendTxOpts() (*bind.TransactOpts, error) {
+    }
+
+    func (m *SimpleTxManager) queryReceipt(ctx context.Context, txID wallet.TxID) *types.Receipt {
+    }
+
+    // estimateGasAndNonce we are explicitly implementing this because
+    // * We want to support legacy transactions (i.e. not dynamic fee)
+    // * We want to support gas management, i.e. add buffer to gas limit
+    func (m *SimpleTxManager) estimateGasAndNonce(ctx context.Context, tx *types.Transaction) (*types.Transaction, error) {
+    }
+    */
+}
+
+#[cfg(test)]
+mod tests {
+    use super::SimpleTxManager;
+    use alloy_consensus::TxLegacy;
+    use alloy_node_bindings::Anvil;
+    use alloy_primitives::{bytes, TxKind::Call, U256};
+    use eigen_logging::{log_level::LogLevel, logger::Logger, tracing_logger::TracingLogger};
+    use once_cell::sync::OnceCell;
+    use tokio;
+
+    static TEST_LOGGER: OnceCell<TracingLogger> = OnceCell::new();
+    const PRIVATE_KEY: &str = "dcf2cbdd171a21c480aa7f53d77f31bb102282b3ff099c78e3118b37348c72f7";
+
+    #[tokio::test]
+    async fn test_send_signed_transaction() {
```

Comment:
can you test the legacy and send_eip1559_tx separately

Reply:
test added